kiwi/system/uri.py

```diff
@@ -134,13 +141,13 @@ def credentials_file_name(self):
 
         :rtype: str
         """
-        uri = urlparse(self.uri)
+        uri = self._has_credentials_query()
         # initialize query with default credentials file name.
         # The information will be overwritten if the uri contains
         # a parameter query with a credentials parameter
         query = {'credentials': 'kiwiRepoCredentials'}
 
-        if uri.query:
+        if uri:
```

Comment:
In fact here is where I got confused while reading the code, had to back to reread uri's assignment.

Reply:
this should be clear now with the method name change ?

Reply:
:+1: